tests/engine/unit/segments/segment_operator.test.ts

```diff
@@ -0,0 +1,25 @@
+import {
+    CONDITION_OPERATORS,
+} from '../../../../flagsmith-engine/segments/constants';
+import {
+    SegmentConditionModel,
+} from '../../../../flagsmith-engine/segments/models';
+import {traitsMatchSegmentCondition} from "../../../../flagsmith-engine/segments/evaluators";
+import {TraitModel} from "../../../../build";
+
+const conditionMatchesTraitkey: [string, string | null | undefined, string | null | undefined, Array<TraitModel > ,boolean][] = [
+    [CONDITION_OPERATORS.IS_SET,'foo',null, [{traitKey:'',traitValue:''}], false],
+    [CONDITION_OPERATORS.IS_SET, 'foo',undefined , [{traitKey:'foo',traitValue:'bar'}], true],
+    [CONDITION_OPERATORS.IS_NOT_SET, 'foo', null,[{traitKey:'foo',traitValue:'bar'}], false],
+    [CONDITION_OPERATORS.IS_NOT_SET, 'foo', undefined, [{traitKey:'',traitValue:''}], true],
+    [CONDITION_OPERATORS.IS_SET, 'foo', null, [{traitKey:'foo',traitValue:'bar'}], true]
+];
+
+test('test_segment_condition_matches_Trait', () => {
+    for (const conditionTraitProperty of conditionMatchesTraitkey) {
```

Comment:
Based on the above, we can then do `for (const testCase of traitExistenceTestCases) {`. I'm not sure what `conditionTraitProperty` means here - it's _really_ important to try and name variables accurately?